src/Disks/S3/DiskS3.cpp

```diff
@@ -127,33 +129,36 @@ void throwIfError(const Aws::Utils::Outcome<Result, Error> & response)
 }
 
 /// Reads data from S3 using stored paths in metadata.
-class ReadIndirectBufferFromS3 final : public ReadIndirectBufferFromRemoteFS<ReadBufferFromS3>
+class ReadIndirectBufferFromS3 final : public ReadBufferFromRemoteFS
```

Comment:
How this one related to `ReadIndirectBufferFromRemoteFS`? This buffer looks direct... Extremely confusing.